File: 867-transpose-matrix/867-transpose-matrix.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& mat) {



        int M = mat.size() , N = mat[0].size();
        vector<vector<int>> ans(N, vector<int>(M,0));
        for (int j =0; j < N; j++)
            for (int i =0 ; i < M; i++)
                ans[j][i] = mat[i][j];

        return ans; 
    }
};

File: 867-transpose-matrix/867-transpose-matrix.py
class Solution:
    def transpose(self, mat: List[List[int]]) -> List[List[int]]:
        
        
        m = len(mat) ; n = len(mat[0])
        r = n ; c = m 
        ans = [ [0]*c for _ in range(r)]
        for i in range(m*n):
            ans[ i %n  ][i//n] = mat[i//n][i%n] 
            
        return ans 
